scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeCompiler.scala

```diff
@@ -381,8 +393,31 @@ class NodeCompiler(
   def compileEnricher(n: Enricher, ctx: ValidationContext, outputVar: OutputVar)(
       implicit nodeId: NodeId,
       scenarioCompilationDependencies: ScenarioCompilationDependencies
-  ): NodeCompilationResult[compiledgraph.service.ServiceRef] = {
-    compileService(n.service, ctx, Some(outputVar))
+  ): NodeCompilationResult[EnricherCompilationResult] = {
+    val serviceCompilationResult = compileService(n.service, ctx, Some(outputVar))
```

Comment:
I think that we should have somethink like
```
(runtimeMode, mockedOutput) match {
  case (Test, Some(...) => compileMockExpression
  case _ => compileService 
}
```
See next comments

Reply:
As discussed we have to compileService to learn the output typing to check mock typing against. 
Service and mock expression is compiled always and decision whether it should be used is actually delegated to `Interpreter`.
In future PRs we can change mock related compilation errors to be treated as warns if we want (like warns about disabled filter nodes).